webpack/scenes/ContentViews/Details/Versions/Compare/CVVersionCompare.js

```diff
@@ -0,0 +1,135 @@
+  const { versionOneId: initialVersionOneId, versionTwoId: initialVersionTwoId } = versionIds;
+  const { versionOneLabel: initialVersionOne, versionTwoLabel: initialVersionTwo } = versionLabels;
+  const { versions: cvDetails } = useSelector(state => selectCVDetails(state, cvId));
+  useEffect(() => {
+    if (!cvDetails) {
+      dispatch(getContentViewDetails(cvId));
+    }
+  }, [dispatch, cvDetails, cvId]);
+  const [versionOne, setVersionOne] = useState(initialVersionOne);
+  const [versionTwo, setVersionTwo] = useState(initialVersionTwo);
+  const getIdFromVersion = useCallback(version => (cvDetails?.find(result =>
+    Number(result.version) === Number(version))?.id), [cvDetails]);
+
+  const versionOneDetails = useSelector(state =>
+    selectCVVersionDetails(state, String(getIdFromVersion(versionOne)
+      ?? initialVersionOneId), cvId));
+  const versionTwoDetails = useSelector(state =>
+    selectCVVersionDetails(state, String(getIdFromVersion(versionTwo)
+      ?? initialVersionTwoId), cvId));
+
+  const tableConfigs = getCVVersionCompareTableConfigs({
+    versionOne,
+    versionTwo,
+    versionOneId: String(getIdFromVersion(versionOne)),
+    versionTwoId: String(getIdFromVersion(versionTwo)),
+  });
+  const filteredTableConfigs = tableConfigs.filter(({ getCountKey }) =>
+    !!getCountKey(versionOneDetails, versionTwoDetails));
+  const versionOneDetailsStatus = useSelector(state =>
+    selectCVVersionDetailsStatus(state, String(getIdFromVersion(versionOne)
+      ?? initialVersionOneId), cvId));
+  const versionTwoDetailsStatus = useSelector(state =>
+    selectCVVersionDetailsStatus(state, String(getIdFromVersion(versionTwo)
+      ?? initialVersionTwoId), cvId));
+
+  useEffect(() => {
+    if (cvDetails) {
+      dispatch(getContentViewVersionDetails(String(getIdFromVersion(versionOne)
+        ?? initialVersionOneId), cvId));
+      dispatch(getContentViewVersionDetails(String(getIdFromVersion(versionTwo)
+        ?? initialVersionTwoId), cvId));
+    }
+  }, [dispatch, versionOne, versionTwo, cvId, getIdFromVersion, cvDetails,
+    initialVersionOneId, initialVersionTwoId]);
+
+
+  const [tab, setTab] = useState('RPM Packages');
+
+  const onSelect = (_e, eventKey) => {
+    // This prevents needless pushing on repeated clicks of a tab
+    if (tab !== eventKey) {
+      setTab(eventKey);
+    }
+  };
+  const currentActiveKey = tab ?? filteredTableConfigs[0].name;
+  const activeTableConfig = filteredTableConfigs.find(config =>
+    String(currentActiveKey) === config.name);
+  const showTabs = filteredTableConfigs.length > 0;
+  return (
+    <>
+      <CVVersionCompareHeader
+        versionOne={versionOne}
+        versionTwo={versionTwo}
+        cvId={cvId}
+        setVersionOne={setVersionOne}
+        setVersionTwo={setVersionTwo}
+      />
+      {showTabs && (((!(versionOneDetailsStatus === STATUS.PENDING
+        ||
+        versionTwoDetailsStatus === STATUS.PENDING || !(versionOne && versionTwo))) &&
```

Comment:
I think we decided to display the CVVesionCompareHeader and have the <Loading/> inside of the table because the last time I talked to @sjha4, he wanted to leave a chance for the users to change the version types in the drop down in case of there is a loading spinner. I hope this makes sense.

Reply:
Yes, I'm just saying the logic here is confusing and hard to read. Can you maybe assign the logic to a variable instead, and then use that here?

So instead of 
```js
(((!(versionOneDetailsStatus === STATUS.PENDING
        ||
        versionTwoDetailsStatus === STATUS.PENDING || !(versionOne && versionTwo))) &&
```
you'd instead do
```js
// earlier in the component, before return
const showCompareTable = (!(versionOneDetailsStatus === STATUS.PENDING
        ||
        versionTwoDetailsStatus === STATUS.PENDING || !(versionOne && versionTwo));

// here
{showTabs && showCompareTable &&
```

Reply:
@jeremylenz sure. I will try to make the logic cleaner. Let me get back to you about it... Thanks